### PR TITLE
chore(dispatcher): remove pipeline mode

### LINKED ISSUE #3308: DispatcherIntegrationTest.java#testPeekFragmentInPipelineMode flaky
**Description**

```
[ERROR] Tests run: 14, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 27.912 s <<< FAILURE! - in io.zeebe.dispatcher.integration.DispatcherIntegrationTest

[ERROR] testPeekFragmentInPipelineMode(io.zeebe.dispatcher.integration.DispatcherIntegrationTest)  Time elapsed: 2.188 s  <<< FAILURE!

java.lang.AssertionError: 



Expecting <AtomicBoolean(true)> to have value:

  <false>

but did not.

	at io.zeebe.dispatcher.integration.DispatcherIntegrationTest.testPeekFragmentInPipelineMode(DispatcherIntegrationTest.java:252)

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

	at java.base/java.lang.reflect.Method.invoke(Method.java:566)

	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)

	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)

	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)

	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)

	at org.junit.rules.RunRules.evaluate(RunRules.java:20)

	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)

	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

	at org.junit.runners.Suite.runChild(Suite.java:128)

	at org.junit.runners.Suite.runChild(Suite.java:27)

	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)

	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)

	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)

	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)

	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)

	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

### PR DESCRIPTION
**Pull request by saig0**
## Description

* remove pipeline mode because it is not used anywhere 

## Related issues

closes #3308 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: dispatcher/src/test/java/io/zeebe/dispatcher/integration/DispatcherIntegrationTest.java [hunk 1]
```diff
@@ -197,180 +196,6 @@ public void run() {
...
-    claimFragment(dispatcher, claimedFragment, totalWork);
-
-    consumerThread1.join();
-    consumerThread2.join();
-
-    dispatcher.close();
-  }
-
-  @Test
-  public void testMarkFragmentAsFailed() throws Exception {
```
Comment (Zelldon) [inline dispatcher/src/test/java/io/zeebe/dispatcher/integration/DispatcherIntegrationTest.java]: Probably we could also remove this feature.

Comment (saig0) [reply dispatcher/src/test/java/io/zeebe/dispatcher/integration/DispatcherIntegrationTest.java]: It seems that it is currently used in transport.

File: dispatcher/src/main/java/io/zeebe/dispatcher/Dispatcher.java [hunk 2]
```diff
@@ -103,6 +97,16 @@ protected void onActorClosing() {
     LOG.debug("Dispatcher closed");
   }
 
+  @Override
+  public void close() {
+    FutureUtil.join(closeAsync());
```
[the remark this answers is not in the document]
Comment (saig0) [reply dispatcher/src/main/java/io/zeebe/dispatcher/Dispatcher.java]: I did not changed it. The method was just moved.

File: dispatcher/src/test/java/io/zeebe/dispatcher/integration/DispatcherIntegrationTest.java (hunk 1 above)
Comment (Zelldon) [reply dispatcher/src/test/java/io/zeebe/dispatcher/integration/DispatcherIntegrationTest.java]: Ah ok thanks

File: dispatcher/src/main/java/io/zeebe/dispatcher/Dispatcher.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zelldon) [reply dispatcher/src/main/java/io/zeebe/dispatcher/Dispatcher.java]: Could you remove it anyway ? :sweat_smile:

**Comment by saig0** [comment]
@Zelldon, I remove the `close()` and `closeAsync()` methods as requested. Please have a look ;)

**Review by ChrisKujawa — APPROVED** [review]
Looks good but I think you broke the build

**Comment by saig0** [comment]
:see_no_evil: yeah, I will fix it

**Comment by saig0** [comment]
bors retry

**Pull request merged**